perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -524,7 +524,10 @@ private void logTotallyRemovedMembers(PerunSession sess, Integer parentGroupId,
 
 	@Override
 	public void deleteAllGroups(PerunSession sess, Vo vo) throws GroupAlreadyRemovedException, GroupAlreadyRemovedFromResourceException, GroupRelationDoesNotExist, GroupRelationCannotBeRemoved {
-		for(Group group: getGroupsManagerImpl().getGroups(sess, vo)) {
+		// get parent groups
+		List<Group> groups = getGroupsManagerImpl().getGroups(sess, vo).stream()
+			.filter(group -> !group.getName().contains(":")).collect(Collectors.toList());
```

Comment:
This is just about conventions, when using Java streams it is a good practice to put each call on a separate line. It improves readability because it is a convention.
```suggestion
		List<Group> groups = getGroupsManagerImpl().getGroups(sess, vo).stream()
			.filter(group -> !group.getName().contains(":"))
			.collect(Collectors.toList());
```